addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/UrlPresenceTestDialog.java

```diff
@@ -0,0 +1,91 @@
+    private static final String ON_FAIL_PARAM = "automation.dialog.urlpresencetest.onfail";
+    private static final String OPERATOR_PARAM = "automation.dialog.urlpresencetest.operator";
+    private static final String REQUEST_HEADER_REGEX =
+            "automation.dialog.urlpresencetest.requestheaderregex";
+    private static final String RESPONSE_HEADER_REGEX =
+            "automation.dialog.urlpresencetest.responseheaderregex";
+    private static final String REQUEST_BODY_REGEX =
+            "automation.dialog.urlpresencetest.requestbodyregex";
+    private static final String RESPONSE_BODY_REGEX =
+            "automation.dialog.urlpresencetest.responsebodyregex";
+
+    private UrlPresenceTest test;
+
+    public UrlPresenceTestDialog(UrlPresenceTest test) {
+        super(
+                View.getSingleton().getMainFrame(),
+                "automation.dialog.urlpresencetest.title",
+                DisplayUtils.getScaledDimension(500, 500));
+        this.test = test;
+        this.addTextField(NAME_PARAM, this.test.getName());
+        this.addNodeSelectField(URL_PARAM, null, true, false);
+        this.addComboField(
+                OPERATOR_PARAM,
+                Arrays.asList(Operator.values()).stream()
+                        .map(Operator::getOperatorValue)
+                        .toArray(String[]::new),
+                test.getData().getOperator());
+        this.addTextField(REQUEST_HEADER_REGEX, this.test.getData().getRequestHeaderRegex());
+        this.addTextField(RESPONSE_HEADER_REGEX, this.test.getData().getResponseHeaderRegex());
+        this.addTextField(REQUEST_BODY_REGEX, this.test.getData().getRequestBodyRegex());
+        this.addTextField(RESPONSE_BODY_REGEX, this.test.getData().getResponseBodyRegex());
+        this.addComboField(
+                ON_FAIL_PARAM,
+                Arrays.asList(OnFail.values()).stream()
+                        .map(OnFail::toString)
+                        .toArray(String[]::new),
+                this.test.getData().getOnFail().toString());
+    }
+
+    @Override
+    public void save() {
+        this.test.getData().setName(this.getStringValue(NAME_PARAM));
+        this.test.getData().setUrl(this.getStringValue(URL_PARAM));
+        this.test.getData().setOperator(this.getStringValue(OPERATOR_PARAM));
+        this.test.getData().setRequestHeaderRegex(this.getStringValue(REQUEST_HEADER_REGEX));
+        this.test.getData().setResponseHeaderRegex(this.getStringValue(RESPONSE_HEADER_REGEX));
+        this.test.getData().setRequestBodyRegex(this.getStringValue(REQUEST_BODY_REGEX));
+        this.test.getData().setResponseBodyRegex(this.getStringValue(RESPONSE_BODY_REGEX));
+        this.test.getData().setOnFail(OnFail.i18nToOnFail(this.getStringValue(ON_FAIL_PARAM)));
+        this.test.getJob().getPlan().setChanged();
+    }
+
+    @Override
+    public String validateFields() {
+        return null;
```

Comment:
This should definitely validate that the regexes are valid, and should probably check the URL is present an valid (unless it includes an env var)

Reply:
Doesnt appear to have been addressed?